sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/AdaptiveSparkPlanExec.scala

```diff
@@ -220,10 +221,28 @@ case class AdaptiveSparkPlanExec(
   }
 
   private def getExecutionId: Option[Long] = {
-    // If the `QueryExecution` does not match the current execution ID, it means the execution ID
-    // belongs to another (parent) query, and we should not call update UI in this query.
     Option(context.session.sparkContext.getLocalProperty(SQLExecution.EXECUTION_ID_KEY))
-      .map(_.toLong).filter(SQLExecution.getQueryExecution(_) eq context.qe)
+      .map(_.toLong)
+  }
+
+  private lazy val shouldUpdatePlan: Boolean = {
+    // If the `QueryExecution` does not match the current execution ID, it means the execution ID
+    // belongs to another (parent) query, and we should call update metrics instead of plan in
+    // this query. For example:
+    //
+    //          ...
+    //           |
+    //  AdaptiveSparkPlanExec (query execution 0, no execution id)
```

Comment:
what does `no execution id` mean?

Reply:
Execution id is assigned by `SQLExecution.withNewExecutionId` and would register to `SQLExecution.executionIdToQueryExecution` and it is used for action. The nested query execution itself is not an action so it does not have execution id.

Reply:
It's a bit hard for me to map this diagram to the code. `getExecutionId` returns the current execution ID, right?

Reply:
yes, but there are two query executions. The current execution ID is mapping to the outer query execution. The code workflow is:

1. assigne a new execution id for an action
2. mapping execution id to outer query execution, so there is no execution id can map to the nesed query execution https://github.com/apache/spark/blob/f8966e7eee1d7f2db7b376d557d5ff6658c80653/sql/core/src/main/scala/org/apache/spark/sql/execution/SQLExecution.scala#L78
3. find the outer query execution and update plan
https://github.com/apache/spark/blob/f8966e7eee1d7f2db7b376d557d5ff6658c80653/sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/AdaptiveSparkPlanExec.scala#L222-L226

Reply:
how about we put it this way: only the root `AdaptiveSparkPlanExec` of the main query that triggers this query execution should update UI.

Reply:
addressed